### PR TITLE
(Feature) #56 User facial record should be delete only after 24hours

### PR DESCRIPTION
**Pull request by yaroslav-fedyshyn-nordwhale**
# Description

About #56 

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/server/verification/verificationAPI.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
 
       try {
         const processor = createEnrollmentProcessor(storage)
-        await processor.enqueueDisposal(enrollmentIdentifier, signature, log)
+        await processor.enqueueDisposal(user, enrollmentIdentifier, signature, log)
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js:43]: why do we need the user record?

Comment (serdiukov-o-nordwhale) [reply src/server/verification/verificationAPI.js:43]: https://github.com/GoodDollar/GoodServer/blob/56-delete_user_facial_record_after_24_hours/src/server/db/mongo/models/delayed-task.js#L22

delayed tasks are linked to the users
for enrollment 24h removal this foreign key isn't used but I've added it to the tasks queue model for the future usage in other cases (as I've projected the queue architecture  more universal)

**Comment by serdiukov-o-nordwhale** [comment]
> Where is 24hours enforced?

https://github.com/GoodDollar/GoodServer/blob/56-delete_user_facial_record_after_24_hours/.env.example#L44

if set to 0, enrollments will be removed immediately (like currently on dev)

File: src/server/verification/verificationAPI.js (hunk 1 above)
Comment (sirpy) [reply src/server/verification/verificationAPI.js:43]: user should be optional.
in the case of faceverification we dont want to keep a link between a user and his facerecord due to privacy

File: src/server/verification/processor/EnrollmentProcessor.js [hunk 2]
```diff
@@ -55,27 +55,40 @@ class EnrollmentProcessor {
...
-    const { provider } = this
+    const { provider, storage } = this
 
-    if (!user || !enrollmentIdenfitier || !payload || !sessionId || !provider.isPayloadValid(payload)) {
+    if (!user || !enrollmentIdentifier || !payload || !sessionId || !provider.isPayloadValid(payload)) {
       throw new Error('Invalid input')
     }
+
+    // make sure user record is not being deleted at the moment
+    const hasDisposalTaskQueued = await storage.hasTasksQueued({ subject: enrollmentIdentifier })
```
Comment (sirpy) [inline src/server/verification/processor/EnrollmentProcessor.js]: missing taskName

File: src/server/verification/processor/EnrollmentProcessor.js [hunk 3]
```diff
@@ -55,27 +55,40 @@ class EnrollmentProcessor {
...
-  async enqueueDisposal(user: any, enrollmentIdentifier, signature, customLogger = noop) {
-    const { storage, provider, keepEnrollments } = this
+  async enqueueDisposal(user: any, enrollmentIdentifier: string, signature: string, customLogger = noop) {
     const recovered = recoverPublickey(signature, enrollmentIdentifier, '')
+    const { storage, provider, adminApi, keepEnrollments } = this
+    const { gdAddress } = user
 
     customLogger.info('Requested disposal for enrollment', { enrollmentIdentifier })
 
+    if (adminApi.isVerified(gdAddress)) {
```
Comment (sirpy) [inline src/server/verification/processor/EnrollmentProcessor.js:93]: need await

File: src/server/verification/processor/EnrollmentProcessor.js (hunk 2 above)
Comment (serdiukov-o-nordwhale) [reply src/server/verification/processor/EnrollmentProcessor.js]: have fixed it in the following commits

**Comment by sirpy** [comment]
@serdiukov-o-nordwhale merge and add env param to heroku QA+DEV
on dev it should delete immediatly on QA after 24hours

**Comment by serdiukov-o-nordwhale** [comment]
@sirpy As it’s large changes pack let’s Yaroslav will also review / re-check locally
Then I’ll merge (in 1-2hrs)

**Comment by yaroslav-fedyshyn-nordwhale** [comment]
Approved

**Pull request merged**